# <center> Анализ вакансий из HH </center>


Загрузка необходимых библиотек:

In [157]:
import psycopg2
import pandas as pd

Соединение с заданными параметрами:

In [158]:
DBNAME = 'project_sql'
USER = 'skillfactory'
PASSWORD = 'cCkxxLVrDE8EbvjueeMedPKt'
HOST = '84.201.134.129'
PORT = 5432

In [159]:

conn = psycopg2.connect(
    dbname=DBNAME,
    user=USER,
    password=PASSWORD,
    host=HOST,
    port=PORT
)
print("Подключение успешно!")

Подключение успешно!


### Задания.

### 3.1. Сколько вакансий есть в базе?

In [160]:
query_3_1=f'''
    select count(*)
    from vacancies
    '''

In [161]:
df=pd.read_sql_query(query_3_1,conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/953150329.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_3_1,conn)


count
0  49197

Количество вакансий в базе составило 49197. 

### 3.2. Сколько работодателей?

In [162]:
query_3_2=f'''
    select count(*)
    from employers
    '''

In [163]:
df=pd.read_sql_query(query_3_2, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/2498192261.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_3_2, conn)


count
0  23501

Количество работодателей в базе составило 23501.

### 3.3. Сколько регионов?

In [164]:
query_3_3=f'''
    select count(*)
    from areas
    '''

In [165]:
df=pd.read_sql_query(query_3_3, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/3822512043.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_3_3, conn)


count
0   1362

Количество регионов составило 1362. 

### 3.4. Сколько сфер деятельности?

In [166]:
query_3_4=f'''
    select count(*)
    from industries
    '''

In [167]:
df=pd.read_sql_query(query_3_4, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/4036238946.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_3_4, conn)


count
0    294

Количество сфер деятельности составило 294.

### 4.1. Напишите запрос, который позволит узнать, сколько (cnt) вакансий в каждом регионе (area).

In [168]:
query_4_1=f'''
    select a.name, count(v.id) cnt
    from areas a
    left join vacancies v ON a.id=v.area_id
    group by 1
    order by 2 DESC
    limit 5
    '''

In [169]:
df=pd.read_sql_query(query_4_1, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/221400649.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_4_1, conn)


name   cnt
0           Москва  5333
1  Санкт-Петербург  2851
2            Минск  2112
3      Новосибирск  2006
4           Алматы  1892

Количество вакансий в каждом регионе составило: 
- Москва	5333
- Санкт-Петербург	2851
- Минск	2112
- Новосибирск	2006
- Алматы	1892.

### 4.2. Посмотрим на зарплаты. У какого количества вакансий заполнено хотя бы одно из двух полей с зарплатой?

In [170]:
query_4_2=f'''
    select count(*)
    from vacancies 
    WHERE (salary_from IS NOT NULL) OR (salary_to IS NOT NULL)
    '''

In [171]:
df=pd.read_sql_query(query_4_2, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/460256265.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_4_2, conn)


count
0  24073

Хотя бы одно из двух полей с зарплатой заполнено у 24073 вакансий. 

### 4.3. Найдите средние значения для нижней и верхней границы зарплатной вилки. Округлите значения до целого числа.

In [172]:
query_4_3=f'''
    select 
    round(AVG(salary_from)) нижняя_граница,
    round(AVG(salary_to)) верхняя_граница
    from vacancies 
    '''

In [173]:
df=pd.read_sql_query(query_4_3, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/27854324.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_4_3, conn)


нижняя_граница  верхняя_граница
0         71065.0         110537.0

Среднее значение для нижней границы зарплатной вилки составило 71065.0, а для верхней границы 110537.0. 

### 4.4. Напишите запрос, который выведет количество вакансий для каждого сочетания типа рабочего графика (schedule) и типа трудоустройства (employment), используемого в вакансиях. Какая пара находится на втором месте по популярности?

In [174]:
query_4_4=f'''
    select schedule, employment, count(*) количество_вакансий
    from vacancies
    group by 1,2
    order by 3 DESC
    offset 1 limit 1
    '''

In [175]:
df=pd.read_sql_query(query_4_4, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/344246506.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_4_4, conn)


schedule        employment  количество_вакансий
0  Удаленная работа  Полная занятость                 7802

Количество вакансий для каждого сочетания типа рабочего графика и типа трудоустройства составило 7802. 

### 4.5. Напишите запрос, выводящий значения поля Требуемый опыт работы (experience) в порядке возрастания количества вакансий, в которых указан данный вариант опыта.

In [176]:
query_4_5=f'''
    select experience, count(*) количество_вакансий
    from vacancies
    group by 1
    order by 2 
    '''

In [177]:
df=pd.read_sql_query(query_4_5, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/698744747.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_4_5, conn)


experience  количество_вакансий
0         Более 6 лет                 1337
1           Нет опыта                 7197
2       От 3 до 6 лет                14511
3  От 1 года до 3 лет                26152

Вывели значение поля "Требуемый опыт работы" в порядке возрастания количества вакансий, в которых указан данный вариант опыта:
- Более 6 лет	1337
- Нет опыта	7197
- От 3 до 6 лет	14511
- От 1 года до 3 лет	26152

### 5.1. Напишите запрос, который позволит узнать, какие работодатели находятся на первом и пятом месте по количеству вакансий.

In [178]:
query_5_1=f'''
    select e.name, count(v.id) количество_вакансий
    from employers e
    left join vacancies v on e.id=v.employer_id
    group by 1
    order by 2 DESC
    limit 5
    '''

In [179]:
df=pd.read_sql_query(query_5_1, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/693413536.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_5_1, conn)


name  количество_вакансий
0         Яндекс                 1933
1     Ростелеком                  491
2       Тинькофф                  444
3           СБЕР                  428
4  Газпром нефть                  331

Написан запрос, который позволил узнать какие работодатели по количеству вакансий находятся на первом и пятом месте:
- на первом месте: Яндекс	1933
- на пятом месте: Газпром нефть	331

### 5.2. Напишите запрос, который для каждого региона выведет количество работодателей и вакансий в нём. Среди регионов, в которых нет вакансий, найдите тот, в котором наибольшее количество работодателей. Впишите его название в поле ниже в том виде, который вернул запрос.

In [180]:
query_5_2=f'''
    select a.name,
           count(e.id) количество_работодателей,
           count(v.id) количество_вакансий
    from areas a
    left join vacancies v on a.id=v.area_id
    left join employers e on a.id = e.area
    group by a.name
    HAVING count(v.id)=0 
    order by 2 DESC
    limit 5
    '''

In [181]:
df=pd.read_sql_query(query_5_2, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/824219045.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_5_2, conn)


name  количество_работодателей  количество_вакансий
0              Россия                       410                    0
1           Казахстан                       207                    0
2  Московская область                        75                    0
3  Краснодарский край                        19                    0
4            Беларусь                        18                    0

Написан запрос, который выводит количество работодателий и вакансий для каждого региона. Среди регионов, в которых нет вакансий, выведен тот, в котором наибольшее количесто работодателей:
- регоион Россия, количество работодателей составляет 410.

### 5.3. Для каждого работодателя посчитайте количество регионов, в которых он публикует свои вакансии.Выберите максимальное значение из получившегося списка.

In [182]:
query_5_3=f'''
    select e.name, count(distinct v.area_id) количество_регионов
    from employers e
    join vacancies v on e.id=v.employer_id
    group by e.id
    order by 2 DESC
    '''

In [183]:
df=pd.read_sql_query(query_5_3, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/3880174466.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_5_3, conn)


name  количество_регионов
0                                      Яндекс                  181
1                                  Ростелеком                  152
2                                  Спецремонт                  116
3                      Поляков Денис Иванович                   88
4                                   ООО ЕФИН                    71
...                                       ...                  ...
14901                            НПП Авиатрон                    1
14902              Центр дистанционных торгов                    1
14903  Городские Телекоммуникационные Системы                    1
14904                       Введенский, Отель                    1
14905              Группа Компаний Автоимпорт                    1

[14906 rows x 2 columns]

Посчитано количество регионов для каждого работодателя, в котрых он публикует свои вакансии. Максимальное значение из списка составляет:
- Яндекс	181

### 5.4. Напишите запрос для подсчёта количества работодателей, у которых не указана сфера деятельности. Введите количество, которое вернул запрос, в поле ниже.

In [184]:
query_5_4=f'''
    select count(*)
    from employers e
    left join employers_industries i on e.id=i.employer_id
    where i.industry_id IS NULL 
    '''

In [185]:
df=pd.read_sql_query(query_5_4, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/3647594158.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_5_4, conn)


count
0   8419

Написан запрос для подсчета количества работодателей, у которых не указана сфера деятельности, их количество составляет 8419.

### 5.5. Напишите запрос, чтобы узнать название компании, находящейся на третьем месте в алфавитном списке (по названию) компаний, у которых указано четыре сферы деятельности. Введите в поле ниже название этой компании так же, как оно указано в результате запроса.

In [186]:
query_5_5=f'''
    select e.name 
    from employers_industries i
    join employers e on e.id=i.employer_id
    group by e.id
    HAVING count(distinct industry_id)=4
    ORDER BY 1 offset 2 limit 1
    '''

In [187]:
df=pd.read_sql_query(query_5_5, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/3293259404.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_5_5, conn)


name
0  2ГИС

Написан запрос, который показывает название компании, находящейся на третьем месте в алфвитном списке (по названию), у которой указано четыре сферы деятельности, компания называется 2ГИС.

### 5.6. С помощью запроса выясните, у какого количества работодателей в качестве сферы деятельности указана «Разработка программного обеспечения».

In [188]:
query_5_6=f'''
    select count(distinct i.employer_id)
    from employers_industries i
    join industries u on i.industry_id=u.id
    where u.name = 'Разработка программного обеспечения'
    '''

In [189]:
df=pd.read_sql_query(query_5_6, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/3370289717.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_5_6, conn)


count
0   3553

В результате запроса можно увидеть количество работодателей, в качестве сферы деятельности которых указана "Разработка програмного обеспечения", их количество составляет 3553. 

### 5.7. Для компании «Яндекс» выведите список регионов-миллионников , в которых представлены вакансии компании, вместе с количеством вакансий в этих регионах. Также добавьте строку Total с общим количеством вакансий компании, собранных в этой таблице. Должна получиться выборка такого вида (приведён пример результата для компании SberTech):

In [190]:
city=('Москва',
'Санкт-Петербург',
'Новосибирск',
'Екатеринбург',
'Казань',
'Красноярск',
'Нижний Новгород',
'Челябинск',
'Уфа',
'Самара',
'Ростов-на-Дону',
'Краснодар',
'Омск',
'Воронеж',
'Пермь',
'Волгоград')

In [191]:
query_5_7=f'''
    (select a.name, count(v.id)
    from vacancies v
    join areas a on a.id=v.area_id
    join employers e on e.id=v.employer_id
    where e.name = 'Яндекс' and a.name in {city}
    group by a.id
    order by 2)
    union all 
    (select 'Total', count(*)
    from vacancies v
    join areas a on a.id=v.area_id
    join employers e on e.id=v.employer_id
    where e.name = 'Яндекс' and a.name in {city})
    '''

In [192]:
df=pd.read_sql_query(query_5_7, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/3020593676.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_5_7, conn)


name  count
0             Total    485
1              Омск     21
2         Челябинск     22
3        Красноярск     23
4         Волгоград     24
5    Ростов-на-Дону     25
6            Казань     25
7             Пермь     25
8            Самара     26
9               Уфа     26
10        Краснодар     30
11          Воронеж     32
12      Новосибирск     35
13  Нижний Новгород     36
14     Екатеринбург     39
15  Санкт-Петербург     42
16           Москва     54

Для компании "Яндекс" выведен список регионов-милионников, в которых представлены вакансии компании, вместе с количеством вакансий в этих регионах. Также добавлена строка Total с общим количеством вакансий компании, собранных в таблице. В результате получилась следующая выборка:
- Total	485
- Омск	21
- Челябинск	22
- Красноярск	23
- Волгоград	24
- Ростов-на-Дону	25
- Казань	25
- Пермь	25
- Самара	26
- Уфа	26
- Краснодар	30
- Воронеж	32
 - Новосибирск	35
- Нижний Новгород	36
- Екатеринбург	39
- Санкт-Петербург	42
- Москва	54

### 6.1. Сколько вакансий имеет отношение к данным?

In [193]:
query_6_1=f'''
select count(id)
from vacancies 
where lower(name) like '%data%' or lower(name) like '%данн%'
'''

In [119]:
df=pd.read_sql_query(query_6_1, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/2450629540.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_6_1, conn)


count
0   1771

К данным имеет отношение количество вакансий в размере 1771.

### 6.2. Сколько есть подходящих вакансий для начинающего дата-сайентиста? Будем считать вакансиями для дата-сайентистов такие, в названии которых есть хотя бы одно из следующих сочетаний:

In [204]:
query_6_2=f'''
select count(id)
from vacancies 
where (lower(name) like '%data scientist%' 
    or lower(name) like '%data science%'
    or lower(name) like '%исследователь данных%' 
    or (name like '%ML%' and lower(name) not like '%html%')
    or lower(name) like '%machine learning%' 
    or lower(name) like '%машинн%обучен%')
and (lower(name) like '%junior%' 
    or experience = 'Нет опыта'
    or employment = 'Стажировка')
'''

In [205]:
df=pd.read_sql_query(query_6_2, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/586518845.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_6_2, conn)


count
0     51

Выведен результат по количеству подходящих вакансий для начинающего дата-сайентиста. Анализ вакансий проводился на основе определенных совпадений по словосочетаниям в названии вакансии. Количество подходящих вакансий составляет 51. 

### 6.3. Сколько есть вакансий для DS, в которых в качестве ключевого навыка указан SQL или postgres? В качестве ответа запишите количество, которое вернул запрос.

In [236]:
query_6_3=f'''
select count(*)
from vacancies 
where (lower(name) like '%data scientist%' 
    or lower(name) like '%data science%'
    or lower(name) like '%исследователь данных%' 
    or (lower (name)like '%ml%' and lower(name) not like '%html%')
    or lower(name) like '%machine learning%' 
    or lower(name) like '%машинн%обучен%')
and (lower(key_skills) like '%sql%'
    or lower(key_skills) like '%postgres%')
'''

In [237]:
df=pd.read_sql_query(query_6_3, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/4053385469.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_6_3, conn)


count
0    229

Проанализировано количество вакансий для DS, в которых в качестве ключевого навыка указаны SQL или postgres. Итоговое количество составляет 229.

### 6.4. С помощью запроса, аналогичного предыдущему, проверьте, насколько популярен Python в требованиях работодателей к DS. Вычислите количество вакансий, в которых в качестве ключевого навыка указан Python. В качестве ответа запишите количество, которое вернул запрос.

In [238]:
query_6_4=f'''
select count(*)
from vacancies 
where (lower(name) like '%data scientist%' 
    or lower(name) like '%data science%'
    or lower(name) like '%исследователь данных%' 
    or (lower (name)like '%ml%' and lower(name) not like '%html%')
    or lower(name) like '%machine learning%' 
    or lower(name) like '%машинн%обучен%')
and (lower(key_skills) like '%python%')
'''

In [239]:
df=pd.read_sql_query(query_6_4, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/2468614928.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_6_4, conn)


count
0    357

С помощью запроса, аналогичного предыдущему, проверена популярность Python в требованиях работодателей к DS. Вчеслено количество вакансий, в которых в качестве ключевого навыка указан Python, их количество составаляет 357.

### 6.5. Сколько ключевых навыков в среднем указывают в вакансиях для DS? Ответ округлите до двух знаков после точки-разделителя.

In [249]:
query_6_5=f'''
select round(avg(length(key_skills)+1 - length(replace(key_skills,chr(9),''))),2)
from vacancies 
where (lower(name) like '%data scientist%' 
    or lower(name) like '%data science%'
    or lower(name) like '%исследователь данных%' 
    or (name like '%ML%' and lower(name) not like '%html%')
    or lower(name) like '%machine learning%' 
    or lower(name) like '%машинн%обучен%')
and (key_skills is not null)
'''

In [250]:
df=pd.read_sql_query(query_6_5, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/2033412393.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_6_5, conn)


round
0   6.41

Среднее количество навыков в вакансиях для DS составляет - 6.41 (ответ округлен до двух знаков после точки-разделителя).

### 6.6. Напишите запрос, позволяющий вычислить, какую зарплату для DS в среднем указывают для каждого типа требуемого опыта (уникальное значение из поля experience). Выясните, на какую зарплату в среднем может рассчитывать дата-сайентист с опытом работы от 3 до 6 лет. Результат округлите до целого числа.

In [257]:
query_6_6=f'''
select experience, round(avg(coalesce((salary_from+salary_to)/2,salary_from,salary_to)),0)
from vacancies 
where (lower(name) like '%data scientist%' 
    or lower(name) like '%data science%'
    or lower(name) like '%исследователь данных%' 
    or (name like '%ML%' and lower(name) not like '%html%')
    or lower(name) like '%machine learning%' 
    or lower(name) like '%машинн%обучен%')
and (key_skills is not null)
group by 1
'''

In [258]:
df=pd.read_sql_query(query_6_6, conn)
df

/var/folders/4p/61221tbs1lq96mmqjjf1m7m40000gn/T/ipykernel_27515/2591831355.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df=pd.read_sql_query(query_6_6, conn)


experience     round
0         Более 6 лет       NaN
1           Нет опыта   74643.0
2  От 1 года до 3 лет  146704.0
3       От 3 до 6 лет  243115.0

Сформирован запрос, который позволяет вычислить, какую запрлату для DS в среднем указывают для каждого типа требуемого опыта. Из чего можно проанализировать на какую зарплату в среднем может расчитывать дата-сайентист с опытом работы от 3 до 6 лет. Такая зарплата составляет 243115.0 (ответ округлен до целого числа).

### Итоговый вывод:
Из проведенного анализа из HH мы узнали следующее, что работодателей меньше, чем вакансий почти в два раза - 23501 к 49197. Это значит, что у каждого работодателя есть по несколько открытых вакансий. 

Среди них самый популярный работодатель по количеству вакансий - Яндекс (1933 вакансии). 
Свои вакансии он публикует в 181 регионах, это максимальное значение из списка работодателей. 
Если делать выборку открытых вакансий Яндекса по регионам-миллионникам, то в совокупности они составляют 485 вакансий, где наибольшее количество в Москве (54),  а наименьшее - в Омске (21).

За основу брали 1362 региона и 294 сфер деятельности. 
Самый популярный регион по вакансиям - Москва (5333). 
Только у половины опубликованных вакансий заполнены поля по зарплате (24073 к 49197). 
Зарплатная вилка в среднем составляет от 71065 до 110537. 
На втором месте по популярности сочетания типа рабочего графика находится удаленная работа и полная занятость (7802). А самый популярный требуемый опыт работы составляет от 1 до 3х лет (26152). 

Также Россия имеет наибольшее количество работодателей, среди которых нет вакансий, их количество составляет 410. 
А также количество работодателей у которых не указана сфера деятельности составляет 8419. 
Также мы узнали, что на третьем месте из списка (в алфавитном порядке) находится компания 2ГИС, у которой указано четыре сферы деятельности. 

Проанализировали, что в качестве сферы деятельности «Разработка программного обеспечения» указана у 3553 работодателей. 
К работе с данными имеет отношение 1771 вакансия. 
Подходящих вакансий для начинающего дата-сайентиста (если делать выборку по определенным словосочетаниям) составляет 51 вакансию. 
В качестве ключевого навыка SQL или Postgres указаны в 229 вакансиях.
Требования к ключевым навыкам Python указаны в 357 вакансиях, это довольно популярное требование. 
В среднем по количеству навыков, которые указаны в вакансиях к DS составляет 6.41. А средняя зарплата для дата-сайентиста с опытом от 3 до 6 лет составляет 243115.0, что является намного выше рынка по зарплате (если отталкиваться от средней ценовой вилки). 